NewInternalsSpecification.html

```diff
@@ -0,0 +1,271 @@
+              1. Let _success_ be the result of calling the [[DefineOwnProperty]] internal method of _newRecord_ passing _N_ and _desc_ as arguments.
+              1. Assert: _success_ is *true*.
+          1. Set the value of the [[Extensible]] internal slot of _newRecord_ to *false*.
+          1. Call EnqueueChangeRecord(_target_, _newRecord_).
+        </emu-alg>
+      </emu-clause>
+    </emu-clause>
+  </emu-clause>
+  <emu-clause id="GetNotifier" aoid="GetNotifier">
+    <h1>GetNotifier(O)</h1>
+    <p>When the abstract operation GetNotifier is called with Object _O_ the following steps are taken:</p>
+    <emu-alg>
+      1. Let _notifier_ be the value of _O_'s [[Notifier]] internal slot.
+      1. If _notifier_ is *undefined*, then
+        1. Let _notifier_ be ObjectCreate(%NotifierPrototype%, ([[Target]], [[ChangeObservers]], [[ActiveChanges]])).
+        1. Set _notifier_'s [[Target]] internal slot to _O_.
+        1. Set _notifier_'s [[ChangeObservers]] internal slot to a new empty List.
+        1. Set _notifier_'s [[ActiveChanges]] internal slot to ObjectCreate(*null*).
+        1. Set _O_'s [[Notifier]] internal slot to _notifier_.
+      1. return _notifier_.
+    </emu-alg>
+  </emu-clause>
+
+  <emu-clause id="BeginChange" aoid="BeginChange">
+    <h1>BeginChange(O, changeType)</h1>
+    <p>When the abstract operation BeginChange is called with Object _O_ and string _changeType_ the following steps are taken:</p>
+    <emu-alg>
+      1. Assert: Type(_O_) is Object.
+      1. Assert: Type(_changeType_) is String.
+      1. Let _notifier_ be GetNotifier(_O_).
+      1. Let _activeChanges_ be the value of _notifier_'s [[ActiveChanges]] internal slot.
+      1. Let _changeCount_ be Get(_activeChanges_, _changeType_).
+      1. ReturnIfAbrupt(_changeCount_).
+      1. If _changeCount_ is *undefined*, let _changeCount_ be 1.
+      1. Else, let _changeCount_ be _changeCount_ + 1.
+      1. Let _success_ be CreateDataProperty(_activeChanges_, _changeType_, _changeCount_).
+      1. Assert: _success_ is *true*.
+    </emu-alg>
+  </emu-clause>
+
+  <emu-clause id="EndChange" aoid="EndChange">
+    <h1>EndChange(O, changeType)</h1>
+    <p>When the abstract operation EndChange is called with Object _O_ and string _changeType_ the following steps are taken:</p>
+    <emu-alg>
+      1. Assert: Type(_O_) is Object.
+      1. Assert: Type(_changeType_) is String.
+      1. Let _notifier_ be GetNotifier(_O_).
+      1. Let _activeChanges_ be the value of _notifier_'s [[ActiveChanges]] internal slot.
+      1. Let _changeCount_ be Get(_activeChanges_, _changeType_).
+      1. ReturnIfAbrupt(_changeCount_).
+      1. Assert: _changeCount_ > 0.
+      1. Let _changeCount_ be _changeCount_ - 1.
+      1. Let _success_ be CreateDataProperty(_activeChanges_, _changeType_, _changeCount_).
+      1. Assert: _success_ is *true*.
+    </emu-alg>
+  </emu-clause>
+
+  <emu-clause id="ShouldDeliverToObserver" aoid="ShouldDeliverToObserver">
```

Comment:
I agree. It's not a good idea to auto-link based on id since most clauses aren't abstract operations. However, I think it's reasonable to say if you have an aoid with an empty value that it will pull from the id.